--- a/gold/target.h.original	2018-08-08 19:53:59.386889220 -0000
+++ b/gold/target.h	2018-08-08 19:59:07.609896824 -0000
@@ -1147,7 +1147,7 @@
   // Record a target-specific program property in the .note.gnu.property
   // section.
   virtual void
-  record_gnu_property(int, int, size_t, const unsigned char*, const Object*)
+  record_gnu_property(int, unsigned int, size_t, const unsigned char*, const Object*)
   { }
 
   // Merge the target-specific program properties from the current object.
--- a/gold/x86_64.cc.original	2018-08-08 19:53:47.394888924 -0000
+++ b/gold/x86_64.cc	2018-08-08 19:58:50.790896409 -0000
@@ -1307,7 +1307,7 @@
   // Record a target-specific program property in the .note.gnu.property
   // section.
   void
-  record_gnu_property(int, int, size_t, const unsigned char*, const Object*);
+  record_gnu_property(int, unsigned int, size_t, const unsigned char*, const Object*);
 
   // Merge the target-specific program properties from the current object.
   void
@@ -1579,7 +1579,7 @@
 template<int size>
 void
 Target_x86_64<size>::record_gnu_property(
-    int, int pr_type,
+    int, unsigned int pr_type,
     size_t pr_datasz, const unsigned char* pr_data,
     const Object* object)
 {
