--- a/js/public/TypeDecls.h	2019-08-28 14:11:03.168076547 +0200
+++ b/js/public/TypeDecls.h	2019-08-28 14:11:54.059077809 +0200
@@ -22,6 +22,7 @@
 
 #include "js-config.h"
 
+#pragma GCC visibility push(default)
 typedef uint8_t jsbytecode;
 
 class JSAtom;
@@ -89,5 +90,6 @@
 typedef PersistentRooted<Value> PersistentRootedValue;
 
 }  // namespace JS
+#pragma GCC visibility pop
 
 #endif /* js_TypeDecls_h */
