--- a/kernel/conftest.sh	2020-06-02 10:30:13.385162009 +0200
+++ b/kernel/conftest.sh	2020-06-02 10:32:18.922167062 +0200
@@ -452,6 +452,8 @@
             # It does not exist on all architectures.
             #
             CODE="
+            #include <asm/page.h>
+            #include <asm/pgtable.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #include <asm/set_memory.h>
             #else
@@ -470,6 +472,8 @@
             # It does not exist on all architectures.
             #
             CODE="
+            #include <asm/page.h>
+            #include <asm/pgtable.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #include <asm/set_memory.h>
             #else
@@ -523,6 +527,8 @@
             # It does not exist on all architectures.
             #
             CODE="
+            #include <asm/page.h>
+            #include <asm/pgtable.h>
             #if defined(NV_ASM_SET_MEMORY_H_PRESENT)
             #include <asm/set_memory.h>
             #else
