--- random/jitterentropy-base.c.orig	2021-09-01 05:44:19.206572363 +0200
+++ random/jitterentropy-base.c	2021-09-01 05:44:26.289571909 +0200
@@ -51,7 +51,12 @@
 
 #undef _FORTIFY_SOURCE
 #ifdef __OPTIMIZE__
-#pragma GCC optimize ("O0")
+  #ifdef __clang__
+    #pragma clang optimize off
+  #else
+    #pragma GCC optimize ("O0")
+  #endif
+  #undef __OPTIMIZE__
 #endif
 
 #include "jitterentropy.h"
