--- a/include/ixion/model_context.hpp	2020-09-23 04:16:20.983101570 +0200
+++ b/include/ixion/model_context.hpp	2020-09-23 04:16:50.652101847 +0200
@@ -61,7 +61,7 @@
         } value;
 
         /** Initializes the cell to be empty. */
-        input_cell(nullptr_t);
+        input_cell(std::nullptr_t);
         /** Boolean cell value. */
         input_cell(bool b);
         /** The char array must be null-terminated. */
