--- a/src/Log.hxx	2021-11-16 08:10:00.845809665 +0100
+++ b/src/Log.hxx	2021-11-16 08:10:24.022808893 +0100
@@ -23,6 +23,7 @@
 #include "LogLevel.hxx"
 #include "util/Compiler.h"
 
+#include <utility>
 #include <exception>
 
 class Domain;
--- a/src/lib/expat/ExpatParser.hxx	2021-11-16 08:14:05.060801537 +0100
+++ b/src/lib/expat/ExpatParser.hxx	2021-11-16 08:14:25.079800870 +0100
@@ -24,6 +24,7 @@
 
 #include <expat.h>
 
+#include <utility>
 #include <stdexcept>
 
 class InputStream;
