--- a/libcxxabi/cmake/Modules/HandleCompilerRT.cmake	2021-10-03 01:52:15.568518323 -0000
+++ b/libcxxabi/cmake/Modules/HandleCompilerRT.cmake	2021-10-03 01:53:03.254520242 -0000
@@ -3,7 +3,7 @@
     message(FATAL_ERROR "LIBCXXABI_COMPILE_FLAGS must be defined when using this function")
   endif()
   set(dest "" PARENT_SCOPE)
-  set(CLANG_COMMAND ${CMAKE_CXX_COMPILER} ${LIBCXXABI_COMPILE_FLAGS}
+  set(CLANG_COMMAND ${CMAKE_CXX_COMPILER} -m${LIBCXX_LIBDIR_SUFFIX} ${LIBCXXABI_COMPILE_FLAGS}
       "--rtlib=compiler-rt" "--print-libgcc-file-name")
   if (CMAKE_CXX_COMPILER_ID MATCHES Clang AND CMAKE_CXX_COMPILER_TARGET)
     list(APPEND CLANG_COMMAND "--target=${CMAKE_CXX_COMPILER_TARGET}")
