--- a/cmake/Modules/HandleCompilerRT.cmake	2018-10-26 19:17:14.955186526 -0000
+++ b/cmake/Modules/HandleCompilerRT.cmake	2018-10-26 19:17:48.613187357 -0000
@@ -3,7 +3,7 @@
     message(FATAL_ERROR "LIBCXX_COMPILE_FLAGS must be defined when using this function")
   endif()
   set(dest "" PARENT_SCOPE)
-  set(CLANG_COMMAND ${CMAKE_CXX_COMPILER} ${LIBCXX_COMPILE_FLAGS}
+  set(CLANG_COMMAND ${CMAKE_CXX_COMPILER} -m${LIBCXX_LIBDIR_SUFFIX} ${LIBCXX_COMPILE_FLAGS}
       "--rtlib=compiler-rt" "--print-libgcc-file-name")
   if (CMAKE_CXX_COMPILER_ID MATCHES Clang AND CMAKE_CXX_COMPILER_TARGET)
     list(APPEND CLANG_COMMAND "--target=${CMAKE_CXX_COMPILER_TARGET}")
