--- test/utils-prng.c.original	2018-08-07 09:04:57.690219651 +0000
+++ test/utils-prng.c	2018-08-07 09:28:33.619254582 +0000
@@ -199,7 +199,7 @@
         }
         else
         {
-#ifdef HAVE_GCC_VECTOR_EXTENSIONS
+#if defined(HAVE_GCC_VECTOR_EXTENSIONS) && defined(__GNUC__) && !defined(__clang__)
             const uint8x16 bswap_shufflemask =
             {
                 3, 2, 1, 0, 7, 6, 5, 4, 11, 10, 9, 8, 15, 14, 13, 12
