--- /src/dos/dos_programs.cpp.original	2018-08-03 00:15:51.567979899 +0200
+++ /src/dos/dos_programs.cpp	2018-08-03 00:16:10.501980366 +0200
@@ -75,7 +75,7 @@
 		for (int d = 0;d < DOS_DRIVES;d++) {
 			if (!Drives[d]) continue;
 
-			char root[7] = {'A'+d,':','\\','*','.','*',0};
+			char root[7] = {static_cast<char>('A'+d),':','\\','*','.','*',0};
 			bool ret = DOS_FindFirst(root,DOS_ATTR_VOLUME);
 			if (ret) {
 				dta.GetResult(name,size,date,time,attr);
--- /src/gui/sdlmain.cpp.original	2018-08-03 00:18:57.848984495 +0200
+++ /src/gui/sdlmain.cpp	2018-08-03 00:19:37.029985461 +0200
@@ -1313,9 +1313,9 @@
 	glUnmapBufferARB = (PFNGLUNMAPBUFFERARBPROC)SDL_GL_GetProcAddress("glUnmapBufferARB");
 	const char * gl_ext = (const char *)glGetString (GL_EXTENSIONS);
 	if(gl_ext && *gl_ext){
-		sdl.opengl.packed_pixel=(strstr(gl_ext,"EXT_packed_pixels") > 0);
-		sdl.opengl.paletted_texture=(strstr(gl_ext,"EXT_paletted_texture") > 0);
-		sdl.opengl.pixel_buffer_object=(strstr(gl_ext,"GL_ARB_pixel_buffer_object") >0 ) &&
+		sdl.opengl.packed_pixel=(strstr(gl_ext,"EXT_packed_pixels") != NULL);
+		sdl.opengl.paletted_texture=(strstr(gl_ext,"EXT_paletted_texture") != NULL);
+		sdl.opengl.pixel_buffer_object=(strstr(gl_ext,"GL_ARB_pixel_buffer_object") != NULL ) &&
 		    glGenBuffersARB && glBindBufferARB && glDeleteBuffersARB && glBufferDataARB &&
 		    glMapBufferARB && glUnmapBufferARB;
     	} else {
--- /src/hardware/serialport/nullmodem.cpp.original	2018-08-03 00:20:43.744987107 +0200
+++ /src/hardware/serialport/nullmodem.cpp	2018-08-03 00:21:10.132987758 +0200
@@ -148,7 +148,7 @@
 	setCTS(dtrrespect||transparent);
 	setDSR(dtrrespect||transparent);
 	setRI(false);
-	setCD(clientsocket > 0); // CD on if connection established
+	setCD(clientsocket != NULL); // CD on if connection established
 }
 
 CNullModem::~CNullModem() {
--- /src/hardware/vga_other.cpp.original	2018-08-03 00:22:22.300989538 +0200
+++ /src/hardware/vga_other.cpp	2018-08-03 00:24:28.132992643 +0200
@@ -280,9 +280,9 @@
 	}
 	Bitu CGApal[4] = {
 		overscan,
-		2 + (color_sel||bw ? 1 : 0) + (background_i ? 8 : 0),
-		4 + (color_sel&&!bw? 1 : 0) + (background_i ? 8 : 0),
-		6 + (color_sel||bw ? 1 : 0) + (background_i ? 8 : 0)
+		static_cast<Bitu>(2 + (color_sel||bw ? 1 : 0) + (background_i ? 8 : 0)),
+		static_cast<Bitu>(4 + (color_sel&&!bw? 1 : 0) + (background_i ? 8 : 0)),
+		static_cast<Bitu>(6 + (color_sel||bw ? 1 : 0) + (background_i ? 8 : 0))
 	};
 	for (Bit8u x=0; x<4; x++) {	 // Position of pixel in question
 		bool even = (x & 1) == 0;
