--- a/compiler-rt/lib/builtins/fp_lib.h	2021-10-03 16:35:49.191823831 -0000
+++ b/compiler-rt/lib/builtins/fp_lib.h	2021-10-03 16:38:21.900818749 -0000
@@ -105,7 +105,7 @@
 COMPILER_RT_ABI fp_t __adddf3(fp_t a, fp_t b);
 
 #elif defined QUAD_PRECISION
-#if __LDBL_MANT_DIG__ == 113 && defined(__SIZEOF_INT128__)
+#if (__LDBL_MANT_DIG__ == 113 || defined(__x86_64__)) && defined(__SIZEOF_INT128__)
 #define CRT_LDBL_128BIT
 typedef uint64_t half_rep_t;
 typedef __uint128_t rep_t;
