--- a/src/inject.cc	2019-08-30 14:58:12.068021524 +0200
+++ b/src/inject.cc	2019-08-30 14:58:31.902022016 +0200
@@ -148,9 +148,10 @@
 
 struct header_field
 {
+  typedef unsigned length_t;
   // member information
   const char* name;
-  unsigned length;
+  length_t length;
   bool is_address;
   bool is_recipient;
   bool is_sender;
@@ -160,7 +161,7 @@
 
   bool present;
 
-  bool parse(mystring& line, bool& rm) 
+  bool parse(mystring& line, bool& rm)
     {
       if(strncasecmp(line.c_str(), name, length))
 	return false;
@@ -204,7 +205,8 @@
 
 #define F false
 #define T true
-#define X(N,IA,IR,IS,IRS,R) { #N ":",strlen(#N ":"),\
+#define X(N,IA,IR,IS,IRS,R) { #N ":", \
+  static_cast<header_field::length_t>(strlen(#N ":")), \
   IA,IR,IS,IRS,R,false, false }
 static header_field header_fields[] = {
   // Sender address fields, in order of priority
@@ -284,14 +286,14 @@
     if(!name) from = user + "@" + host;
     else      from = user + "@" + host + " (" + name + ")";
   }
-  
+
   mystring suser = getenv("NULLMAILER_SUSER");
   if(!suser) suser = user;
 
   mystring shost = getenv("NULLMAILER_SHOST");
   if(!shost) shost = host;
   canonicalize(shost);
-  
+
   if(use_header_sender && !sender)
     sender = suser + "@" + shost;
 }
