--- a/src/Experimental.h	2019-08-30 21:14:01.620580652 +0200
+++ b/src/Experimental.h	2019-08-30 21:14:05.030580736 +0200
@@ -148,7 +148,9 @@
 // RBD, 1 Sep 2008
 // Enables MIDI Output of NoteTrack (MIDI) data during playback
 // USE_MIDI must be defined in order for EXPERIMENTAL_MIDI_OUT to work
+#ifdef USE_MIDI
 #define EXPERIMENTAL_MIDI_OUT
+#endif
 // JKC, 17 Aug 2017
 // Enables the MIDI note stretching feature, which currently
 // a) Is broken on Linux (Bug 1646)
