--- a/lib-src/FileDialog/Makefile.in	2018-10-23 20:24:04.226274026 -0000
+++ b/lib-src/FileDialog/Makefile.in	2018-10-23 20:25:28.933276116 -0000
@@ -216,7 +216,7 @@
 am__v_OBJCXX_0 = @echo "  OBJCXX  " $@;
 am__v_OBJCXX_1 = 
 OBJCXXLD = $(OBJCXX)
-OBJCXXLINK = $(LIBTOOL) $(AM_V_lt) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
+OBJCXXLINK = $(LIBTOOL) $(AM_V_lt) --tag CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=link $(OBJCXXLD) $(AM_OBJCXXFLAGS) $(OBJCXXFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
 AM_V_OBJCXXLD = $(am__v_OBJCXXLD_@AM_V@)
@@ -536,7 +536,7 @@
 	win/$(DEPDIR)/$(am__dirstamp)
 
 libFileDialog.la: $(libFileDialog_la_OBJECTS) $(libFileDialog_la_DEPENDENCIES) $(EXTRA_libFileDialog_la_DEPENDENCIES) 
-	$(AM_V_OBJCXXLD)$(OBJCXXLINK) -rpath $(libdir) $(libFileDialog_la_OBJECTS) $(libFileDialog_la_LIBADD) $(LIBS)
+	$(AM_V_OBJCXXLD)$(OBJCXXLINK) --tag CXX -rpath $(libdir) $(libFileDialog_la_OBJECTS) $(libFileDialog_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
