From 54daff18aabcdc4a36dab8623e74f9f3e9742455 Mon Sep 17 00:00:00 2001
From: David Carlos Manuelda <StormByte@gmail.com>
Date: Tue, 7 Aug 2018 05:11:45 +0200
Subject: [PATCH] Fix compilation with CLang

---
 Make.defaults | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Make.defaults b/Make.defaults
index c4c0592..700b08c 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -35,7 +35,7 @@ RANLIB	:= $(CROSS_COMPILE)gcc-ranlib
 PKGS	=
 
 SUBDIR_CFLAGS ?=
-clang_cflags =
+clang_cflags = `${PKG_CONFIG} --cflags efivar`
 gcc_cflags =
 cflags	= $(EXTRALIBDIR) $(EXTRAINCDIR) $(CFLAGS) $(SUBDIR_CFLAGS) \
 	-Werror -Wall -Wextra -Wsign-compare -Wstrict-aliasing \
