--- a/panel-plugin/sensors-plugin.c	2021-09-30 15:24:17.886964190 +0200
+++ b/panel-plugin/sensors-plugin.c	2021-09-30 15:25:05.032961163 +0200
@@ -1014,7 +1014,7 @@
     sensors->widget_sensors = gtk_box_new ((sensors->plugin_mode == XFCE_PANEL_PLUGIN_MODE_HORIZONTAL) ? GTK_ORIENTATION_HORIZONTAL : GTK_ORIENTATION_VERTICAL, 0);
 
     /* initialize value label widget */
-    sensors->panel_label_text = gtk_widget_new (GTK_TYPE_LABEL, "label", _("<span><b>Sensors</b></span>"), "use-markup", TRUE, "xalign", 0.0, "yalign", 0.5, "margin", INNER_BORDER, NULL);
+    sensors->panel_label_text = gtk_widget_new (GTK_TYPE_LABEL, "label", "", "use-markup", TRUE, "xalign", 0.0, "yalign", 0.5, "margin", INNER_BORDER, NULL);
 
     gtk_widget_show (sensors->panel_label_text);
 
